### PR TITLE
Dark mode redux

### PR DESCRIPTION
**Pull request by broskoski**
Screen Shot 2020-05-20 at 1 33 27 PM
Screen Shot 2020-05-20 at 1 33 38 PM
Screen Shot 2020-05-20 at 1 34 09 PM

### CODE REVIEW

**Comment by broskoski** [comment]
In case it's not obvious, this turns dark mode into an actual theme.

File: src/v2/components/Billing/components/MyGroups/components/MyGroup/components/UpgradeSelection/index.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const OptionLabel = styled(Text).attrs({
 
 const Option = styled(Box)`
   &:hover ${OptionLabel} {
-    color: black;
+    color: ${props => props.theme.colors.gray.bold};
```
Comment (dzucconi) [inline src/v2/components/Billing/components/MyGroups/components/MyGroup/components/UpgradeSelection/index.js:31]: Incase you wanna save yourself some typing in the future: https://github.com/styled-system/styled-system/tree/master/packages/theme-get

File: src/v2/components/UI/GenericButton/index.js [hunk 2]
```diff
@@ -55,14 +65,30 @@ export const buttonBorderWidth = x =>
...
+      props.theme.name === 'default'
+        ? translucentGray('bold')
+        : translucentWhite('bold')};
+  color: ${props =>
+    props.theme.name === 'default'
+      ? translucentGray('bold')
+      : translucentWhite('bold')};
 `
 
+// To-do switch function based on theme
```
Comment (dzucconi) [inline src/v2/components/UI/GenericButton/index.js:81]: If you want `TODO`s to show up in ur VSCode use the format `TODO: Foo`

File: src/v2/components/UI/Layouts/BlankLayout/components/LegacyDarkTheme/index.js [hunk 3]
```diff
@@ -12,39 +12,5 @@ export const toggle = mode => {
...
-      }
-    }
-
-    // Firefox
-    @-moz-document url-prefix() {
-      body {
-        background-color: black;
-      }
-    }
-  }
```
Comment (dzucconi) [inline src/v2/components/UI/Layouts/BlankLayout/components/LegacyDarkTheme/index.js:49]: 🙏

**Comment by dzucconi** [comment]
Very exciting @broskoski 💡 🕶️

**Pull request merged**